### PR TITLE
new case: nwfilter-binding-list check

### PR DESCRIPTION
**Pull request by juteman**
automation about nwfilter-binding-list

Signed-off-by: qiulzhan <qiulzhan@redhat.com>

### CODE REVIEW

**Comment by juteman** [comment]
```
JOB ID     : 9eaaf73d701ad1f4a1ae3d89b932786ac5985ecb
JOB LOG    : /root/avocado/job-results/job-2018-08-23T17.09-9eaaf73/job.log
 (1/1) type_specific.io-github-autotest-libvirt.nwfilter_binding_list.variable_notation: PASS (36.23 s)

```

File: libvirt/tests/src/nwfilter/nwfilter_binding_list.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    new_filter_2 = params.get("newfilter_2")
+    time_wait = params.get("time_wait", 10)
+    option = params.get("option")
+    vmxml_backup = libvirt_xml.vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    # prepare vm filterrfer parameters dict list
+    filter_param_list_1 = []
+    params_key_1 = []
+    filter_param_list_2 = []
+    params_key_2 = []
+    for i in params.keys():
```
Comment (chloerh) [inline libvirt/tests/src/nwfilter/nwfilter_binding_list.py]: params_key_1 = [i for i in params.keys() if 'parameters_name_' in i]

File: libvirt/tests/src/nwfilter/nwfilter_binding_list.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    for i in range(len(params_key_2)):
+        params_dict = {}
+        params_dict['name'] = params[params_key_2[i]]
+        params_dict['value'] = params['dhcp_value_%s' % i]
+        filter_param_list_2.append(params_dict)
+    filterref_dict_2 = {}
+    filterref_dict_2['name'] = new_filter_2
+    filterref_dict_2['parameters'] = filter_param_list_2
+
+    def env_setting(filterref_dict_1, filterref_dict_2):
```
Comment (kylazhang) [inline libvirt/tests/src/nwfilter/nwfilter_binding_list.py]: Need function description and parameters explanation

File: libvirt/tests/src/nwfilter/nwfilter_binding_list.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        new_filterref = new_iface_2.new_filterref(**filterref_dict_2)
+        new_iface_2.filterref = new_filterref
+        logging.debug("new interface xml is: %s \n %s" %
+                      (new_iface_1, new_iface_2))
+        vmxml.add_device(new_iface_1)
+        vmxml.add_device(new_iface_2)
+        vmxml.sync()
+        return new_iface_1, new_iface_2
+
+    def attach_new_device():
```
Comment (kylazhang) [inline libvirt/tests/src/nwfilter/nwfilter_binding_list.py]: As above

File: libvirt/tests/src/nwfilter/nwfilter_binding_list.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        newnet_iface = interface.Interface('network')
+        newnet_iface.source = {'network': "default"}
+        filterref_dict = {}
+        filterref_list = [{'name': "CTRL_IP_LEARNING", 'value': "dhcp"}]
```
Comment (kylazhang) [inline libvirt/tests/src/nwfilter/nwfilter_binding_list.py]: Hard code please move to cfg

File: libvirt/tests/src/nwfilter/nwfilter_binding_list.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        # list binding port dev
+        ret = virsh.nwfilter_binding_list(debug=True)
+        utlv.check_exit_status(ret, status_error)
+        # detach a interface
+        option = "--type network" + " --mac " + new_iface_1.mac_address
+        time.sleep(time_wait)
+        ret = virsh.detach_interface(vm_name, option, debug=True)
+        utlv.check_exit_status(ret, status_error)
+        ret = virsh.nwfilter_binding_list(debug=True)
+        utlv.check_exit_status(ret, status_error)
```
Comment (kylazhang) [inline libvirt/tests/src/nwfilter/nwfilter_binding_list.py]: Still no list result check?

File: libvirt/tests/src/nwfilter/nwfilter_binding_list.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        # set env of start vm
+        new_iface_1, new_iface_2 = env_setting(
+            filterref_dict_1, filterref_dict_2)
+        # start vm
+        virsh.start(vm_name, debug=True)
+        # list binding port dev
+        ret = virsh.nwfilter_binding_list(debug=True)
+        utlv.check_exit_status(ret, status_error)
+        # detach a interface
+        option = "--type network" + " --mac " + new_iface_1.mac_address
```
Comment (kylazhang) [inline libvirt/tests/src/nwfilter/nwfilter_binding_list.py]: It can also be set in cfg

**Comment by kylazhang** [comment]
Merge the PR as he left and any update will be submitted later

**Pull request merged**